File: reverse.java
class reverse {
    public static void main(String[] args) {
        int n = Integer.parseInt(args[0]);
        System.out.printf("%d -> %d\n", n, reverse(n));
    }
    static int reverse(int n) {
        int m = 0;
        int i = digit(n) - 1;
        for(; i >= 0; i--) {
            m += (n % 10) * Math.pow(10, i);
            n /= 10;
        }
        return m;
    }
    static int digit(int n) {
        return (int) Math.ceil(Math.log10(n));
    }
}

File: reverse.c
//usr/bin/env make -s "${0%.c}" && ./"${0%.c}" "$@"; s=$?; rm ./"${0%.c}"; exit $s

#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int digits(int n) {
    return ceil(log10(n));
}

int reverse(int n) {
    int m = 0;
    for(int i = digits(n) - 1; i >= 0; i--) {
        div_t d = div(n, 10);
        n = d.quot;
        m += d.rem * pow(10, i);
    }
    return m;
}

int main(int argc, char *argv[]) {
    int n = atoi(argv[1]);
    printf("%d -> %d\n", n, reverse(n));
}
